packages/@ember/-internals/glimmer/lib/modifiers/on.ts

```diff
@@ -0,0 +1,241 @@
+import { assert } from '@ember/debug';
+import { DEBUG } from '@glimmer/env';
+import { Opaque, Simple } from '@glimmer/interfaces';
+import { Tag } from '@glimmer/reference';
+import { Arguments, CapturedArguments, ModifierManager } from '@glimmer/runtime';
+import { Destroyable } from '@glimmer/util';
+
+/*
+  Internet Explorer 11 does not support `once` and also does not support
+  passing `eventOptions`. In some situations it then throws a weird script
+  error, like:
+
+  ```
+  Could not complete the operation due to error 80020101
+  ```
+
+  This flag determines, whether `{ once: true }` and thus also event options in
+  general are supported.
+*/
+const SUPPORTS_EVENT_OPTIONS = (() => {
+  try {
+    const div = document.createElement('div');
+    let counter = 0;
+    div.addEventListener('click', () => counter++, { once: true });
+
+    let event;
+    if (typeof Event === 'function') {
+      event = new Event('click');
+    } else {
+      event = document.createEvent('Event');
+      event.initEvent('click', true, true);
+    }
+
+    div.dispatchEvent(event);
+    div.dispatchEvent(event);
+
+    return counter === 1;
+  } catch (error) {
+    return false;
+  }
+})();
+
+export class OnModifierState {
+  public tag: Tag;
+  public element: Element;
+  public args: CapturedArguments;
+  public eventName!: string;
+  public callback!: EventListener;
+  private userProvidedCallback!: EventListener;
+  public once?: boolean;
+  public passive?: boolean;
+  public capture?: boolean;
+  public options?: AddEventListenerOptions;
+  public shouldUpdate = true;
+
+  constructor(element: Element, args: CapturedArguments) {
+    this.element = element;
+    this.args = args;
+    this.tag = args.tag;
+  }
+
+  updateFromArgs() {
+    let { args } = this;
+
+    let { once, passive, capture }: AddEventListenerOptions = args.named.value();
```

Comment:
In `ember-on-modifier`, we assert that only these three options are passed to guard against typos. Do you think we should add this here as well?

Reply:
Hmm, the RFC specifically states the opposite (that passing other things is "ignored"):

From [here](https://emberjs.github.io/rfcs/0471-on-modifier.html#named-parameters):

> The following named parameters will be accepted as options to {{on}}:
> 
> * capture
> * once
> * passive
> 
> These will be passed forward to addEventListener as options in modern browsers, and will be polyfilled in older browsers. Other options will be ignored.
> 

I don't feel super strongly so I just followed the RFC on this.

@pzuraq - Do you think we should change to assert?